### PR TITLE
Add TextMate contribution documentation

### PR DESCRIPTION
**Pull request by marechal-p**
Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

### CODE REVIEW

File: doc/SUMMARY.md [hunk 1]
```diff
@@ -9,9 +9,11 @@
 * [Services and Contributions](Services_and_Contributions.md)
 
 ## Using Theia
+
 * [Build your own IDE](Composing_Applications.md)
 * [Authoring an Extension](Authoring_Extensions.md)
 * [Authoring Plug-ins](Authoring_Plugins.md)
+* [TextMate Support](TextMate.md)
```
Comment (simark) [inline doc/SUMMARY.md]: I find the title a bit ambiguous.  TextMate is a text editor, we don't support running TextMate in Theia :).  Also, it's not clear what this even means for somebody that doesn't know what TextMate grammars are and that you can use them for syntax highlighting.  So I would prefer something like "Syntax highlighting using TextMate grammars".

File: doc/TextMate.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# TextMate Support in Theia
+
+TextMate grammars allow us to colorize most source files with accurate
+decorations, even if it is only at the syntactic level (no language deep
+understanding). Semantic coloring can be provided by Language Servers, among
```
Comment (simark) [inline doc/TextMate.md]: I don't think semantic coloring is part of the LSP (yet).

File: doc/SUMMARY.md (hunk 1 above)
Comment (marechal-p) [reply doc/SUMMARY.md]: `TextMate Coloring`?

File: doc/TextMate.md (hunk 2 above)
Comment (marechal-p) [reply doc/TextMate.md]: No but I felt like it was worth mentioning, to give a bit of context.

File: doc/SUMMARY.md [hunk 3]
```diff
@@ -9,9 +9,11 @@
 * [Services and Contributions](Services_and_Contributions.md)
 
 ## Using Theia
+
 * [Build your own IDE](Composing_Applications.md)
 * [Authoring an Extension](Authoring_Extensions.md)
 * [Authoring Plug-ins](Authoring_Plugins.md)
+* [TextMate Coloring](TextMate.md)
```
Comment (simark) [inline doc/SUMMARY.md]: TextMate Syntax Coloring?

File: doc/TextMate.md (hunk 2 above)
Comment (svenefftinge) [reply doc/TextMate.md]: Theia supports a provisional extension. So I think it is fair to mention it like that.

File: doc/SUMMARY.md (hunk 3 above)
Comment (svenefftinge) [reply doc/SUMMARY.md]: I think it should be 'Adding Language Support'. 
And then 'the grammar (textmate) would be one part.
Configuration and language server two others.

**Pull request merged**